dbt_subprojects/dex/models/bot_trades/banana_gun/base/banana_gun_base_bot_trades.sql

```diff
@@ -0,0 +1,193 @@
+      botTrades
+    GROUP BY
+      tx_hash
+  ),
+  botETHDeposits AS (
+    SELECT
+      tx_hash,
+      block_number,
+      CAST(value AS DECIMAL (38, 0)) AS deltaGwei,
+      CAST(value AS DECIMAL (38, 0)) AS depositGwei
+    FROM
+      {{ source('base','traces') }}
+      JOIN botContracts ON to = botContracts.address
+    WHERE
+      {% if is_incremental() %}
+      {{ incremental_predicate('block_time') }}
+      {% else %}
+      block_time >= TIMESTAMP '{{project_start_date}}'
+      {% endif %}
+      AND value > CAST(0 AS UINT256)
+  ),
+  botETHWithdrawals AS (
+    SELECT
+      tx_hash,
+      block_number,
+      CAST(value AS DECIMAL (38, 0)) * -1 AS deltaGwei,
+      0 AS depositGwei,
+      block_hash,
+      to
+    FROM
+      {{ source('base','traces') }}
+      JOIN botContracts ON "from" = botContracts.address
+    WHERE
+      {% if is_incremental() %}
+      {{ incremental_predicate('block_time') }}
+      {% else %}
+      block_time >= TIMESTAMP '{{project_start_date}}'
+      {% endif %}
+      AND value > CAST(0 AS UINT256)
+  ),
+  botEthTransfers AS (
+    /* Deposits */
+    (
+      SELECT
+        tx_hash,
+        block_number,
+        deltaGwei,
+        depositGwei
+      FROM
+        botETHDeposits
+    )
+    UNION ALL
+    /* Withdrawals */
+    (
+      SELECT
+        tx_hash,
+        block_number,
+        deltaGwei,
+        depositGwei
+      FROM
+        botETHWithdrawals
+    )
+  ),
+  botEthDeltas AS (
+    SELECT
+      tx_hash,
+      block_number,
+      SUM(deltaGwei) AS feeGwei,
+      SUM(depositGwei) AS depositGwei
+    FROM
+      botEthTransfers
+    GROUP BY
+      tx_hash,
+      block_number
+  )
+SELECT
+  block_time,
+  date_trunc('day', botTrades.block_time) as block_date,
+  date_trunc('month', botTrades.block_time) as block_month,
+  '{{project_name}}' as bot,
+  block_number,
+  '{{blockchain}}' AS blockchain,
+  -- Trade
+  amount_usd,
+  type,
+  token_bought_amount,
+  token_bought_symbol,
+  token_bought_address,
+  token_sold_amount,
+  token_sold_symbol,
+  token_sold_address,
+  -- Fees
+  ROUND(
+    CAST(feeGwei AS DOUBLE) / CAST(depositGwei AS DOUBLE),
+    /* Round feePercentage to 0.01% steps */
+    4
+  ) AS fee_percentage_fraction,
+  (feeGwei / 1e18) * price AS fee_usd,
+  feeGwei / 1e18 fee_token_amount,
+  '{{fee_token_symbol}}' AS fee_token_symbol,
+  {{weth}} AS fee_token_address,
+  -- Dex
+  project,
+  version,
+  token_pair,
+  project_contract_address,
+  -- User
+  user,
+  botTrades.tx_hash,
+  evt_index,
+  IF(evt_index = highestEventIndex, true, false) AS is_last_trade_in_transaction
+FROM
+  botTrades
+  JOIN highestEventIndexForEachTrade ON botTrades.tx_hash = highestEventIndexForEachTrade.tx_hash
+  LEFT JOIN botETHDeltas ON botTrades.tx_hash = botETHDeltas.tx_hash
+  LEFT JOIN {{ source('prices', 'usd') }} ON (
+    blockchain = '{{blockchain}}'
+    AND contract_address = {{weth}}
+    AND minute = DATE_TRUNC('minute', block_time)
```

Comment:
should add an incremental block here to filter prices on the incremental run

Reply:
how would that look like in a join condition? it already has minute = minute, is that not sufficient?

Reply:
```suggestion
  LEFT JOIN {{ source('prices', 'usd') }} ON (
    blockchain = '{{blockchain}}'
    AND contract_address = {{weth}}
    AND minute = DATE_TRUNC('minute', block_time)
    {% if is_incremental() %}
    AND {{ incremental_predicate('minute') }}
    {% endif %}  
```

Reply:
while indeed we are already filtering on minute, adding the incremental predicate adds a static filter which is more efficient and reduces the total amount of rows that the engine need to scan to look for the exact minute match

Reply:
For these models, the performance difference is negligible but we try to be consistent in adding incremental filters to all sources/refs when possible.